hle58-01.py

```diff
@@ -139,24 +193,50 @@ def run_petri_net():
         messageHandler("Το προϊόν προστέθηκε:")
         canRunTransition0 = False
 
-    if transition1():
+    if transition1() or transition11():
         canvas.itemconfigure(token_TK_P10, state="normal")
         tokenHandler("token_TK_P10")
-
-        # buy_product_button.config(state="active")
-        # cancel_product_button.config(state="active")
         changeAmount = calculateChangeHandler()
-        messageHandler(f"Εισαγωγή νομίσματος: €{coin_var.get()}, το ποσό συμπληρώθηκε, έδωσε ρέστα €{changeAmount}")
+        messageHandler(f"Εισαγωγή νομίσματος: €{coin_var.get()}, το ποσό συμπληρώθηκε, έδωσε ρέστα €{changeAmount}")        
 
     if transition2():
         canvas.itemconfigure(token_TK_P4, state="normal")
         tokenHandler("token_TK_P4")
         remainingAmount = remainingValueHandler()
         messageHandler(f"Εισαγωγή νομίσματος: €{coin_var.get()}, υπολείπονται ακόμη €{remainingAmount}")
+        cancel_product_button.config(state="active")
```

Comment:
Maybe we could add cancel_product_button.config(state="active") only on transition0 which is the first step when the user chooses a product, and going forward will be able to cancel on any other transition.